Scripts/Player.gd

```diff
@@ -45,6 +45,14 @@ var closest_interactable: Node3D
 @export var stamina_threshold: float = 0.45
 var stamina_regenerating = false
 
+@export var blink_timer: float = 1.0
+@export var max_blink: float = 1.0
+@export var blink_deplete_rate: float = 0.1
+var blink_current_deplete_rate = blink_deplete_rate
+@onready var blink_overlay: ColorRect = $UI/Blink
```

Comment:
Player class shouldn't be responsible for UI. Ideally, create a separate node that listens for player's signals and changes it's child controls accordingly.